### DISCUSSION TITLE
Sub events + Timecode integration

### DISCUSSION THREAD

**Discussion by hbilney** [cpvalente/ontime#1218] [Ideas]
It would be cool to use events as acts on stage but also be able to add 'sub events' to an event for cues in different departments (lighting, video, automation etc). Currently I can only find a way to have one 'event' active at once.

It would also be nice if each parent 'event' could have a timecode offset and children events within the parent could be triggered with an LTC/Midi/Artnet timecode feed.

These features would be game changing for live music events!

**Comment by cpvalente** [comment]
Hi @hbilney, thank you for taking the time to reach out.

I postponed replying to you until I had a chance to chat with other maintainers.

**Sub events**
We have plans to create sub events in Ontime, this will likely mean a data restructure and we are waiting for the new version to stabilise.
However, we dont currently have plans to allow having parallel running events. Perhaps you could share more details on the use case and workflow?

This is likely the issue we will use to track the feature https://github.com/cpvalente/ontime/issues/204

**Timecode in**
We have tested this a few times, and there was even a beta release of Ontime that consumed timecode 
However we found that it opens a lot of questions and generally didnt play well with the rest of the application structure:

To allow triggering events from timecode, we would need to consume globally in the app and decide what does that input do.
- How would the user activate this? a special time of roll mode? or a per event trigger?
- what precedence does this take over the remaining playback?

If we have a compelling argument or use case for this I am happy to reconsider. You can check the previous conversations in the previous issue https://github.com/cpvalente/ontime/issues/535
